lib/good_job/active_job_extensions/concurrency.rb

```diff
@@ -40,6 +43,9 @@ module Concurrency
           key = job.good_job_concurrency_key
           next if key.blank?
 
+          # Don't attempt to enforce concurrency limits with other queue adapters.
+          next unless job.queue_adapter.is_a?(ActiveJob::QueueAdapters::GoodJobAdapter)
```

Comment:
I think this should be the first guard clause inside of the `before_perform` block.